### PR TITLE
Tp

### PR DESCRIPTION
**Pull request by nick-drozd**

### CODE REVIEW

File: protos/txn_payload.proto [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// -----------------------------------------------------------------------------
+
+syntax = "proto3";
+
+message OMITransactionPayload {
+    string action = 1;
+
+    bytes object = 2;
```
Comment (peterschwarz) [inline protos/txn_payload.proto]: This should be `payload`or `data`.

File: omi/sawtooth_omi/handler.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    def apply(self, transaction, state):
+        action, txn_obj, signer = _unpack_transaction(transaction)
+
+        address = make_omi_address(txn_obj, action)
+
+        state_obj = _get_state_object(state, address, action)
+
+        if state_obj:
+            _validate_object(state_obj, action, signer)
```
Comment (peterschwarz) [inline omi/sawtooth_omi/handler.py]: This is less about validating the object, which there should probably be some (minimal, at the moment) validation of the transaction payload, and more about authorizing updates.

I.e. if not `state_obj`, it's an insert, else update.

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: omi/sawtooth_omi/handler.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
```
Comment (jsmitchell) [inline omi/sawtooth_omi/handler.py]: Need to do validation of the txn_obj here.

**Review by jsmitchell — CHANGES_REQUESTED** [review]

File: omi/sawtooth_omi/handler.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+from sawtooth_omi.protobuf.work_pb2 import Work
+from sawtooth_omi.protobuf.recording_pb2 import Recording
+from sawtooth_omi.protobuf.identity_pb2 import IndividualIdentity, OrganizationalIdentity
+from sawtooth_omi.protobuf.txn_payload_pb2 import OMITransactionPayload
+
+
+# actions
+
+WORK = 'SetWork'
```
Comment (jsmitchell) [inline omi/sawtooth_omi/handler.py]: I would recommend you just use the action name. By taking the verb out, we'll run into trouble if we add alternative actions on the same objects.

Comment (nick-drozd) [reply omi/sawtooth_omi/handler.py]: Right now the action doubles as a type tag that gets carried around with the object, so it's convenient to use variables. I'll come up with something cleaner later.

Comment (peterschwarz) [reply omi/sawtooth_omi/handler.py]: Doesn't hurt to separate those out.  In fact, if you removed them, you probably don't need constants for the `SetX` actions.

File: omi/sawtooth_omi/handler.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+    '''
+    header = TransactionHeader()
+    header.ParseFromString(transaction.header)
+    signer = header.signer_pubkey
+
+    payload = OMITransactionPayload()
+    payload.ParseFromString(transaction.payload)
+
+    action = payload.action
+    txn_obj = transaction.object
```
Comment (peterschwarz) [inline omi/sawtooth_omi/handler.py]: This hasn't been updated for the `object` -> `data` change

**Review by peterschwarz — APPROVED** [review]

**Review by jsmitchell — APPROVED** [review]

**Pull request merged**